app/js/sumo_db.js

```diff
@@ -154,6 +153,21 @@
       endpoint += '?format=json'; // TODO bug 1088014
 
       return request(endpoint, 'GET').then(JSON.parse);
+    },
+
+    update_preference: function(user, setting) {
+      var endpoint = API_V2_BASE + 'user/';
+      endpoint += user.username + '/';
+
+      var delete_setting = endpoint + 'delete_setting/';
+      delete_setting += '?format=json'; // TODO bug 1088014
+      return request_with_auth(delete_setting, 'POST', setting)
+        .then(function(response) {
```

Comment:
Yes, I also have a temp workaround for the 404 on delete. Let me add then, and you can have a look. Seeing I already did it for the metadata PR, it's not really extra work from my side.